pylint/typing.py

```diff
@@ -2,11 +2,48 @@
 # For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
 
 """A collection of typing utilities."""
-from typing import TYPE_CHECKING, Dict, List, Union
+import sys
+from typing import TYPE_CHECKING, Dict, List, NamedTuple, Union
 
 if TYPE_CHECKING:
     from typing import Counter  # typing.Counter added in Python 3.6.1
 
+if sys.version_info >= (3, 8):
+    from typing import Literal, TypedDict
+else:
+    from typing_extensions import Literal, TypedDict
+
+
+class FileItem(NamedTuple):
+    """Represents data about a file handled by pylint
+
+    Each file item has:
+    - name: full name of the module
+    - filepath: path of the file
+    - modname: module name"""
```

Comment:
```suggestion
    - modname: module name
    """
```
PEP8 -> closing quotes should be on a new line if docstring > 1 line.